pom.xml

```diff
@@ -141,7 +141,7 @@ flexible messaging model and an intuitive client API.</description>
     <!-- apache commons -->
     <commons-compress.version>1.15</commons-compress.version>
 
-    <bookkeeper.version>4.7.2</bookkeeper.version>
+    <bookkeeper.version>4.9.0-20181031-b17b165-SNAPSHOT</bookkeeper.version>
```

Comment:
why are we depending on SNAPSHOT version? is it possible to update the bk-version once we actually have official release.? and depending changes can be tested on individual's test branch.

Reply:
yes, I had the concern that SNAPSHOT version gets  removed after sometime and it's  temporary so, pulsar pipeline might fail at that time. also, SNAPSHOT might not be considered as stable version right? so, is it possible to avoid using such SNAPSHOT versions?

Reply:
>  I had the concern that SNAPSHOT version gets removed after sometime and it's temporary so, pulsar pipeline might fail at that time. 

I am not sure I understand this. why pipeline will fail, if the version is pinned to a gitsha.

> also, SNAPSHOT might not be considered as stable version right? 

sure. but master in pulsar is also "snapshot", no? the point is for pulsar to take bk's version as early as possible, so that when bk and pulsar are released, they are all tested together for a long time. since 2.2.0 is just released, we have time for 2.3.0 to be released. why not try to bump bk version in the early stage of 2.3.0, rather than late in the 2.3.0 release cycle.

Reply:
> why pipeline will fail, if the version is pinned to a gitsha.

yes, I was not aware about it. generally SNAPSHOT version considered as temporary and maven-repo expires them after sometime.

Reply:
a snapshot version only expires when a version is released. for example, `4.8.0-SNAPSHOT` will be expired when 4.8.0 is released. 

but since we attached gitsha information, it will have two benefits: 1) the version will not be changed; 2) the version will not be expired (you can check https://repository.apache.org/content/repositories/snapshots/org/apache/bookkeeper/bookkeeper-server/ , as we have a few snapshots published with gitsha, they are not auto-expired). 

does this address your concern? or do you really need an official release here?

Reply:
> a snapshot version only expires when a version is released. for example, 4.8.0-SNAPSHOT will be expired when 4.8.0 is released.

sure.